sqlgen/db.go

```diff
@@ -361,6 +361,52 @@ func (db *DB) InsertRow(ctx context.Context, row interface{}) (sql.Result, error
 	return db.execWithTrace(ctx, query, "InsertRow")
 }
 
+// InsertRows inserts multiple rows into the database, chunksize rows at a time.
+// Most SQL db enforce a limit on max size of packet, which is why we need to break
+// the rows into chunks.
+//
+// rows should be an array of pointers to a struct, for example:
+//
+//   user1 := &User{Name: "foo"}
+//   user2 := &User{Name: "fan"}
+//   if err := db.InsertRows(ctx, []{interface{}}{user1, user2}, 100); err != nil {
+//
+func (db *DB) InsertRows(ctx context.Context, rows []interface{}, chunkSize int) (sql.Result, error) {
+	ctx, tx, err := db.WithTx(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	var result sql.Result
+	for j := 0; j < len(rows); j += chunkSize {
+		sliceLength := chunkSize
+		if len(rows) < j+chunkSize {
+			sliceLength = len(rows) - j
+		}
+		slice := rows[j:sliceLength]
+
+		query, err := db.Schema.MakeBatchInsertRow(slice)
+		if err != nil {
+			return nil, err
+		}
+
+		numColumns := len(query.Columns)
+		for i := 0; i < len(slice); i++ {
+			if err := db.checkColumnValuesAgainstLimits(ctx, query, query.Columns, query.Values[i*numColumns:(i+1)*numColumns], query.Table); err != nil {
+				return nil, err
+			}
+		}
+		result, err = db.execWithTrace(ctx, query, "InsertRows")
```

Comment:
I'm not sure what's the implication of calling execWithTrace many times using the same operation name.

Reply:
Also, it might be weird to return partial `result`. Maybe we don't return result at all?

Reply:
> I'm not sure what's the implication of calling execWithTrace many times using the same operation name.

@changpingc We already do that when someone calls InsertRow in a loop, right ? So nothing different here

Reply:
> Also, it might be weird to return partial result. Maybe we don't return result at all?

Makes sense